### PR TITLE
User Story 5 - create recipe

### PR DESCRIPTION
**Pull request by stekug**

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
 /** @type {import('next').NextConfig} */
 const nextConfig = {
+  typescript: {
+    ignoreBuildErrors: false,
```
Comment (NikRoe) [inline next.config.js:4]: This should be the default, so I think you can remove this setting

File: pages/_app.tsx [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
+  const [recipes, setRecipes] = useLocalStorageState('recipes', {
+    defaultValue: recipesJson,
+  });
+
+  const handleAddRecipe = (recipe: Recipe) => {
+    setRecipes((currData) => [recipe, ...currData]);
+  };
+
   const handleToggleFavorite: HandleToggleFavoriteFunction = (id) => {
     const favoriteSet = new Set(favoriteRecipesList);
```
Comment (NikRoe) [inline pages/_app.tsx]:
```suggestion
    const favoriteSet = new Set<string>(favoriteRecipesList);
```

File: pages/_app.tsx [hunk 3]
```diff
@@ -1,16 +1,24 @@
...
+import recipesJson from '@/lib/recipes.json';
 import type { AppProps } from 'next/app';
-import { HandleToggleFavoriteFunction } from '@/types';
+import { HandleToggleFavoriteFunction, Recipe } from '@/types';
 
 export default function App({ Component, pageProps }: AppProps) {
   const [favoriteRecipesList, setFavoriteRecipesList] = useLocalStorageState(
     'favoriteRecipesList',
     { defaultValue: [] }
   );
```
Comment (NikRoe) [inline pages/_app.tsx]:
```suggestion
  const [favoriteRecipesList, setFavoriteRecipesList] = useLocalStorageState<string[]>(
    'favoriteRecipesList',
    { defaultValue: [] }
  );
```

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi Team,

nicely done 🙌 

I noticed that in your `tsconfig.json` the strict mode for TypeScript is still set to `false`. Please set it to `true`.
After setting it to `true`, I had to make a few changes to some of your files:

In your `_app.tsx` you need to type your favorite State:
```js
const [favoriteRecipesList, setFavoriteRecipesList] = useLocalStorageState<string[]>('favoriteRecipesList', { defaultValue: [] });
```

And also to your `Set`:

```js
const favoriteSet = new Set<string>(favoriteRecipesList);
```

I also changed your `AppBaseProps` to be this:
```js
  favoriteRecipes?: Recipe[];
  recipes?: Recipe[];
  onToggleFavorite: (id: string) => {};
  favoriteRecipesList: string[];
```

In your `pages/index.tsx` and your `pages/favorites.tsx` your recipe prop can look like this to cover the potential case where it is `undefined`: `recipes={recipes || []}`

Also in your `pages/favorites.tsx` you can do the same for the `favoriteRecipesList`:  `favoriteRecipesList={favoriteRecipesList || []}`

I noticed that each recipe has a description and an instructions array with objects that also contain a key description. This looks to me like the description is stored twice. If that is the case, you should remove the one you don't need.

The texts in your input fields for the recipe name and the ingredient name would benefit a lot from some padding-left.

Moreover, I think that having almost only controlled components in your form is not really needed. Since you don't need the information entered in an input at that very moment, but only on form submission. Therefore, I think modifying all the data makes sense in the handleSubmit function.
But this will require a lot of refactoring and for the moment, I want you to focus on the suggestions I gave on the PR and also in this comment and to continue working on your other features after the suggestions have been implemented.
In the meantime, I will try to refactor this form for you and once that is done, we'll meet to implement it together on your end.

Keep it up 🙌

File: next.config.js (hunk 1 above)
Comment (gunnarmon) [reply next.config.js:4]: we just added this for development, so we don't run into build errors.

File: pages/_app.tsx (hunk 3 above)
Comment (gunnarmon) [reply pages/_app.tsx]: thanks!

File: pages/_app.tsx (hunk 2 above)
Comment (gunnarmon) [reply pages/_app.tsx]: thanks!

**Review by NikRoe — APPROVED** [review]
👍

**Comment by gunnarmon** [comment]
we also added handling for error cases when our recipes data and recipeFavoriteList data is undefined.

**Pull request merged**